source/agora/node/Config.d

```diff
@@ -170,15 +175,74 @@ public struct InterfaceConfig
      /// Bind address
     public string address;
 
-    /// Bind port
-    public ushort port;
+    /***************************************************************************
+
+        Bind port, default to the scheme's default
+
+        Default values are 80 for `http`, 443 for `https`, and 2826 for `tcp`.
+
+    ***************************************************************************/
+
+    public @Optional ushort port;
+
+    /***************************************************************************
+
+        The public address of this interface
+
+        If this binds to an interface that is not publicly available,
+        this value needs to be set for the node to correctly register itself
+        on the registry.
+
+    ***************************************************************************/
+
+    public string hostname;
 
     /// Default values when none is given in the config file
     private static immutable InterfaceConfig[Type.max] Default = [
         // Publicly enabled by default
-        { type: Type.http, address: "0.0.0.0", port: 0xB0A, },
-        { type: Type.tcp,  address: "0.0.0.0", port: 0xA0B, },
+        { type: Type.tcp,  address: "0.0.0.0", port: 0xB0A, },
+        { type: Type.http, address: "0.0.0.0", port: 8080, },
     ];
+
+    ///
+    public void validate () const
+    {
+        auto slashIdx = this.hostname.indexOf('/');
+        ensure(slashIdx < 0 || this.type == Type.http || this.type == Type.https,
+               "Cannot have path ('{}') in with an interface of type '{}': Remove '{}' or use http(s)",
+               this.hostname, this.type, this.hostname[slashIdx .. $]);
+
+        const host = this.hostname[0 .. (slashIdx < 0) ? $ : slashIdx];
+        ensure(isValidHostName(host),  "'{}' is not a valid hostname name / IP address", host);
+    }
+
+    /// Get the address that this interface can be registered as
+    public Address getAddress () const
+    {
+        // Assume validation already happened
+
+        auto slashIdx = this.hostname.indexOf('/');
+        if (slashIdx < 0)
+            slashIdx = this.hostname.length;
+        const host = this.hostname[0 .. slashIdx];
+        const path = this.hostname[slashIdx .. $];
+
+        string schema;
+        final switch (this.type)
+        {
+        case Type.http:
+            schema = "http";
+            break;
+        case Type.https:
+            schema = "https";
+            break;
+        case Type.tcp:
+            schema = "agora";
+            break;
+        }
+
+        return Address(URL(schema, host, this.port, InetPath(path)));
```

Comment:
This limits user to use of same port on its machine and public address

Reply:
We should probably fuzzy parse `hostname` (or whatever we rename it to) and patch it as needed.
So if one wants to have a different port for their public address, they can just give it in `hostname`.

Reply:
Issue also mentions `register`.